### ISSUE TITLE
Implement Pytorch Metrics

### ISSUE THREAD

**Issue by bonham79** [CUNY-CL/yoyodyne#158]
TorchMetrics support is pretty reliable nowadays and makes distributed training less annoying (no more World sizes, yay!). It also syncs well with Wandb logging and allows monitoring of training batch performance. Any complaints about me migrating validation logging to this?

**Comment by kylebgorman** [comment]
I don't know anything about it but it sounds like a positive.

**Comment by bonham79** [comment]
It's just a suite of metrics that torch sets up to manage multi-gpus under the hood. You just pass it your tensors during the training loop and it will store submetrics. Then when you need the actual metric you call and it does the calculation and memory collection under the hood. It saves you from desync issues if you have more than GPU for training. Also PTL supports it so it helps reduce boilerplate for other metrics.

**Comment by Adamits** [comment]
I think I tried to do this last year, and had some issues getting the features I actually wanted from it to track, so I gave up, but I think it was in beta or something then. I'd be happy if you got it working :)

**Comment by bonham79** [comment]
I just got it running for BLEU scores on another project, so CER/WER should be stable by now.

**Comment by kylebgorman** [comment]
[I piloted this a bit])https://colab.research.google.com/drive/1XJBEovJ-uQozr3kspC3RsPZF_hKoZqBp?usp=sharing). We can do our form of accuracy using `torchmetrics.classification.MulticlassExactMatch`, but they don't have an equivalent of our SER, just edit distance limited to strings. (So we'd have to convert back to strings, add all the necessary bookkeeping of letting the models see the indexes, and we wouldn't really be doing "SER" anymore either.) [I filed a feature request](https://github.com/Lightning-AI/torchmetrics/issues/2856) but probably the best solution is to just use the same API they do to implement SER rather than submitting it to their library.

**Comment by bonham79** [comment]
It's somewhat painless to implement new metrics so long as the distributed training is managed properly. I support implementing it on our side and then once it's robust enough we can push to library.
